### PR TITLE
Support building on OpenBSD, x86-64

### PR DESCRIPTION
**Pull request by alex-segura**
I was able to get maiko to compile, and medley running (with ./run-medley) with this patch.

### CODE REVIEW

File: bin/makefile-openbsd.x86_64-x [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+# This is to make the %$#@! Apollo cc happy
+OEXT = .o
+# OPTFLAGS is normally -O2.
+OPTFLAGS =  -O2 -g3
+DISPOPTFLAGS = -O2 -g3
+FPFLAGS =
+DFLAGS = -DOPENBSD -DAIX -DOLD_CURSOR \
+	-DBYTESWAP -DFORKCOMM -DNOFORN -DLOGINT  $(XFLAGS) \
+	-DRELEASE=351 -DSYSVSIGNALS
```
Comment (waywardmonkeys) [inline bin/makefile-openbsd.x86_64-x:28]: `SYSVSIGNALS` is no longer needed here.

File: bin/makefile-openbsd.x86_64-x [hunk 2]
```diff
@@ -0,0 +1,54 @@
+# Options for OpenBSD, Intel x86_64 and X-Window
+
+#CC = gcc -m64 $(GCC_CFLAGS) -fno-omit-frame-pointer -Wall -Wextra -fno-aggressive-loop-optimizations
+CLANG_CFLAGS += -I/usr/X11R6/include
```
Comment (waywardmonkeys) [inline bin/makefile-openbsd.x86_64-x:4]: This can go in `XFLAGS` below.

File: bin/makefile-openbsd.x86_64-x [hunk 3]
```diff
@@ -0,0 +1,54 @@
+# Options for OpenBSD, Intel x86_64 and X-Window
+
+#CC = gcc -m64 $(GCC_CFLAGS) -fno-omit-frame-pointer -Wall -Wextra -fno-aggressive-loop-optimizations
```
Comment (waywardmonkeys) [inline bin/makefile-openbsd.x86_64-x:3]: The flags after the `GCC_FLAGS` can be removed. I have an open PR removing them from other files.

**Review by waywardmonkeys — CHANGES_REQUESTED** [review]
Looks good to me with a couple of minor things.

You might also give the cmake build a quick update.

**Review by nbriggs — COMMENTED** [review]
I second waywardmonkeys' notes.

**Comment by alex-segura** [comment]
Updated in consideration of @waywardmonkeys comments.

File: bin/makefile-openbsd.x86_64-x [hunk 4]
```diff
@@ -1,8 +1,6 @@
 # Options for OpenBSD, Intel x86_64 and X-Window
 
-#CC = gcc -m64 $(GCC_CFLAGS) -fno-omit-frame-pointer -Wall -Wextra -fno-aggressive-loop-optimizations
-CLANG_CFLAGS += -I/usr/X11R6/include
-CC = clang -m64 $(CLANG_CFLAGS)
+CC = gcc -m64 $(GCC_CFLAGS)
```
Comment (nbriggs) [inline bin/makefile-openbsd.x86_64-x:3]: Isn't Clang the default compiler on OpenBSD these days?  I'd pick whichever one is installed by default while choosing clang over gcc if they're both there.

Comment (alex-segura) [reply bin/makefile-openbsd.x86_64-x:3]: Yes, that's probably the better way to go.

**Review by nbriggs — APPROVED** [review]
LGTM.  @waywardmonkeys  ?

**Review by waywardmonkeys — APPROVED** [review]
LGTM.

**Comment by alex-segura** [comment]
Awesome, thanks for working with me to get this merged.

**Pull request merged**